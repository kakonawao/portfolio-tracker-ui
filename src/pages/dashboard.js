import React from "react";
import { connect } from "react-redux";

import AccountsSection from "../components/accountsSection";
import TransactionsSection from "../components/transactionsSection";
import SessionManager from "../components/sessionManager";


const mapStateToProps = (state) => {
    return {
        session: state.session
    }
};

class DashboardPage extends React.Component {

    render() {
        if (this.props.session.authenticated) {
            return (
                <div>
                    <AccountsSection/>
                    <TransactionsSection/>
                </div>
            );
        }

        return (
            <div>
                <SessionManager />
            </div>
        );
    }
}

export default connect(mapStateToProps)(DashboardPage);
