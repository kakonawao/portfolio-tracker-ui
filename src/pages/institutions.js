import React from "react";


export class InstitutionsPage extends React.Component {

    render() {
        return (
            <h2>Institutions</h2>
        )
    }
}

export default InstitutionsPage;
