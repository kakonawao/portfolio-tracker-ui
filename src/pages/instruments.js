import React from "react";


export class InstrumentsPage extends React.Component {

    render() {
        return (
            <h2>Instruments</h2>
        )
    }
}

export default InstrumentsPage;
