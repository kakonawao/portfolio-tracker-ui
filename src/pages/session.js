import React from "react";
import { connect } from "react-redux";

import LoginForm from "../components/loginForm";
import { login } from "../actions/session";


const mapStateToProps = (state) => {
    return {
        authenticated: state.session.authenticated,
        username: state.session.username
    }
};

export class SessionPage extends React.Component {

    constructor(props) {
        super(props);

        this.inputs = {
            username: React.createRef(),
            password: React.createRef()
        }
    }

    login = (event) => {
        event.preventDefault();
        const loginData = {
            username: this.inputs.username.current.value,
            password: this.inputs.password.current.value
        };
        this.props.dispatch(login(loginData));
    };

    renderUserData() {
        return (
            <div>
                <h2>Session</h2>
                <p>You are logged in as {this.props.username}.</p>
            </div>
        )
    }

    renderLoginForm() {
        return (
            <div>
                <h2>Session</h2>
                <LoginForm login={this.login}
                           username={this.inputs.username}
                           password={this.inputs.password}/>
            </div>
        );
    }

    render() {
        if (this.props.authenticated) {
            return this.renderUserData();
        }

        return this.renderLoginForm();
    };
}

export default connect(mapStateToProps)(SessionPage);
