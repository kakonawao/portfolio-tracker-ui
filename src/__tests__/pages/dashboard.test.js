import React from "react";
import { shallow } from "enzyme";

import { DashboardPage } from "../../pages/dashboard";


describe("<DashboardPage>", () => {

    it("renders a button that redirects to session page is user is not authenticated", () => {
        const historyPush = jest.fn();
        const props = {
            dispatch: jest.fn(),
            history: {
                push: historyPush
            },
            session: {
                authenticated: false
            }
        };

        const wrapper = shallow(<DashboardPage {...props}/>);

        // Check snapshot
        expect(wrapper).toMatchSnapshot();

        // Click button and check that history redirected to /session
        wrapper.find('button').simulate("click");
        expect(props.history.push.mock.calls.length).toEqual(1);
        expect(props.history.push.mock.calls[0][0]).toEqual("/session");
    });

    it("renders accounts and transactions components when user is authenticated", () => {
        const props = {
            dispatch: jest.fn(),
            session: {
                authenticated: true
            }
        };

        const wrapper = shallow(<DashboardPage {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })
});
