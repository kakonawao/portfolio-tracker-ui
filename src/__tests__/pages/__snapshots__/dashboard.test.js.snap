// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage> renders a button that redirects to session page is user is not authenticated 1`] = `
<div>
  <h2>
    Login Required
  </h2>
  <p>
    You need to log in first.
  </p>
  <button
    onClick={[Function]}
  >
    Log in
  </button>
</div>
`;

exports[`<DashboardPage> renders accounts and transactions components when user is authenticated 1`] = `
<div>
  <Connect(AccountsSection) />
  <TransactionsSection />
</div>
`;
